app/src/main/java/fr/free/nrw/commons/utils/SystemThemeUtils.java

```diff
@@ -0,0 +1,51 @@
+package fr.free.nrw.commons.utils;
+
+import android.content.Context;
+import android.content.res.Configuration;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.kvstore.JsonKvStore;
+import fr.free.nrw.commons.settings.Prefs;
+
+public class SystemThemeUtils {
+
+    private Context context;
+
+    @Inject
+    @Named("default_preferences")
+    JsonKvStore applicationKvStore;
```

Comment:
You shouldn't perform field injection and constructor injection, wherever possible use constructor injection. Also this will not be injected and should crash the application with a NPE.

Move `@Named("default_preferences") JsonKvStore applicationKvStore` into the constructor like context

Reply:
Done! Thanks for the suggestion.